include/net/nrf_cloud_log.h

```diff
@@ -0,0 +1,117 @@
+ * @{
+ */
+
+/**
+ * @brief Initialize the nRF Cloud logging subsystem. Call after
+ * the date_time library is ready.
+ */
+void nrf_cloud_log_init(void);
+
+/**
+ * @brief Set log_level to the level at or below which a log message will be
+ * sent to the cloud.
+ *
+ * Log messages with a log level larger than this will not be sent.
+ * For example, if log_level is LOG_LEVEL_WRN (2), then
+ * LOG_LEVEL_WRN (2) and LOG_LEVEL_ERR (1) messages will be sent, and
+ * LOG_LEVEL_INF (3) and LOG_LEVEL_DBG (4) will not.
+ * Set log_level to LOG_LEVEL_NONE (0) to disable logging.
+ * See zephyr/include/zephyr/logging/log_core.h.
+ *
+ * @param log_level The log level to go to the cloud.
+ */
+void nrf_cloud_log_control_set(int log_level);
+
+/**
+ * @brief Get current log level to be sent to nRF Cloud.
+ *
+ * @retval Current log level, if CONFIG_NRF_CLOUD_LOG_BACKEND is enabled, otherwise
+ *         0, indicating logs are disabled.
+ */
+int nrf_cloud_log_control_get(void);
+
+/**
+ * @brief Tell REST-based logger the REST context and device_id for later
+ *        use when outputting a log. This is used by the nRF Cloud logs backend
+ *        when CONFIG_NRF_CLOUD_LOG_BACKEND=y.
+ *
+ * @param[in] ctx REST context to use.
+ * @param[in] dev_id Device ID to send message on behalf of.
+ */
+void nrf_cloud_log_rest_context_set(struct nrf_cloud_rest_context *ctx, const char *dev_id);
+
+/**
+ * @brief Enable or disable logging to the cloud.
+ * @param[in] enable Set true to send logs to the cloud, false to disable.
+ */
+void nrf_cloud_log_enable(bool enable);
+
+/**
+ * @brief Find out if logger is enabled.
+ *
+ * @retval True if logger is enabled.
+ */
+bool nrf_cloud_log_is_enabled(void);
+
+#if defined(CONFIG_NRF_CLOUD_LOG_DIRECT)
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+/**
+ * @brief Directly log to the cloud. This does not use the Zephyr logging
+ * system if CONFIG_NRF_CLOUD_LOG_BACKEND is disabled. Otherwise, it is passed to
+ * the logging system.
+ *
+ * @param log_level The log level for this message; dropped if above set log level.
+ * @param fmt Format string.
+ * @param ... Format arguments.
+ * @return 0 if log line is successfully sent, otherwise, a negative error number.
+ */
+int nrf_cloud_log_send(int log_level, const char *fmt, ...);
+
+#elif defined(CONFIG_NRF_CLOUD_REST)
+/**
+ * @brief Directly log to the cloud. This does not use the Zephyr logging
+ * system if CONFIG_NRF_CLOUD_LOG_BACKEND is disabled. Otherwise, it is passed to
+ * the logging system.
+ *
+ * @param ctx REST interface context to use to send log message.
+ * @param dev_id Device id to use when sending the log message via REST.
+ * @param log_level The log level for this message; dropped if above set log level.
+ * @param fmt Format string.
+ * @param ... Format arguments.
+ * @return 0 if log line is successfully sent, otherwise, a negative error number.
+ */
+int nrf_cloud_rest_log_send(struct nrf_cloud_rest_context *ctx, const char *dev_id,
```

Comment:
Check whether the prefix of the function should be aligned with `nrf_cloud_log_rest_context_set()`